repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -725,6 +725,10 @@ private Object mapVertexToAttribute(AtlasVertex entityVertex, AtlasAttribute att
         List<AtlasObjectId> ret        = null;
         List                softRefVal = entityVertex.getListProperty(attribute.getVertexPropertyName(), List.class);
 
+        if (CollectionUtils.isEmpty(softRefVal)) {
+            return softRefVal;
+        }
+
         if (CollectionUtils.isNotEmpty(softRefVal)) {
```

Comment:
replace line #732 with: "else {", as the reverse condition is already tested in line #728 above.